blockonomics.php

```diff
@@ -556,42 +560,79 @@ public function displayForm()
+                                            {
+                                                $("#advance-settings-toggle").text("Hide Advanced Settings");
+                                            } else 
+                                            {
+                                                $("#advance-settings-toggle").text("Show Advanced Settings");
+                                            }
+                                          });
+                                        });
+                                      </script>
+                                      <button type="button" id="advance-settings-toggle" class="btn btn-default btn-lg">
+                                        Show Advanced Settings
+                                      </button>',
+              ),                
+              array(
+                  'type' => 'select',
+                  'label' => $this->l('Time Period'),
+                  'name' => 'BLOCKONOMICS_TIMEPERIOD',
+                  'desc' => $this->l('Countdown timer on payment page'),
+                  'required' => false,
+                  'id' => 'advance-settings-1',
+                  'form_group_class' => 'hide',
+                  'options' => array(
+                      'query' => array(
+                          array('key' => '10', 'name' => $this->l('10 minutes')),
+                          array('key' => '15', 'name' => $this->l('15 minutes')),
+                          array('key' => '20', 'name' => $this->l('20 minutes')),
+                          array('key' => '25', 'name' => $this->l('25 minutes')),
+                          array('key' => '30', 'name' => $this->l('30 minutes')),
+                      ),
+                      'id' => 'key',
+                      'name' => 'name',
+                  ),
+              ),
+              array(
+                  'type' => 'text',
+                  'label' => $this->l('Pay by bitcoin icon size'),
+                  'desc' => $this->l(
+                      'Size in pixels.
+                       Set 0 to disable icon'
+                  ),
+                  'name' => 'BLOCKONOMICS_LOGO_HEIGHT',
+                  'required' => false,
+                  'class' => 'fixed-width-xl',
+                  'form_group_class' => 'hide',
+                  'id' => 'advance-settings-2',
+              ),
+              array(
+                  'type' => 'html',
+                  'label' => $this->l('Underpayment Slack %'),
+                  'desc' => $this->l('Allow payments that are off by a small percentage'),
+                  'required' => false,
+                  'form_group_class' => 'hide',
+                  'html_content' => '<input type="number" class="fixed-width-xl" id="advance-settings-3" min=0 max=10 step=0.01 name="BLOCKONOMICS_UNDERPAYMENT_SLACK" value=' . strval($slack_value) . '>',
```

Comment:
min and max values do not halt the form submission. This may lead to confusion or undesired behavior if the user inputs a negative value. Lets validate this value is correct in updateSettings and also increase max value for underpayment slack to 20% to match woocommerce future changes:
https://github.com/blockonomics/woocommerce-plugin/issues/246